plugins/org.jkiss.dbeaver.ext.postgresql.ui/src/org/jkiss/dbeaver/ext/postgresql/ui/editors/PostgreSourceViewEditor.java

```diff
@@ -82,6 +98,32 @@ public void run() {
                         refreshPart(PostgreSourceViewEditor.this, true);
                     }
                 }, true));
+            contributionManager.add(ActionUtils.makeActionContribution(
+                new Action(PostgreMessages.procedure_check_label, Action.AS_PUSH_BUTTON) {
+                    {
+                        setToolTipText(PostgreMessages.procedure_check_description);
+                    }
+                        
+                    @Override
+                    public void run() {
+                        IWorkbenchWindow workbenchWindow = UIUtils.getActiveWorkbenchWindow();
+                        List<DBSProcedure> entities = new ArrayList<>();
+                        entities.add((DBSProcedure) sourceObject);
+                        DBRRunnableWithResult<String> generator = SQLEditorHandlerCheckProcedureConsole.checkGenerator(entities);
+                        UIUtils.runInUI(workbenchWindow, generator);
+                        String sql = CommonUtils.notEmpty(generator.getResult());
+                        SQLNavigatorContext navContext = new SQLNavigatorContext(sourceObject);
+                        String procName = ((DBSProcedure) sourceObject).getName();
+                        String title = procName + " " + PostgreMessages.procedure_check_label2;
```

Comment:
I'm not a fan of concatenating localized strings. Please use `Check {0}` and `NLS.bind`.